wd_demo.au3

```diff
@@ -200,10 +202,9 @@ Func RunDemo($idDebugging, $idBrowsers, $idUpdate, $idHeadless)
 				EndIf
 			Next
 
-			_WD_DeleteSession($sSession)
-
 		EndIf
 
+		_WD_DeleteSession($sSession)
```

Comment:
I'm failing to see how this is an improvement. Do you intend to call `_WD_DeleteSession` even when `_WD_CreateSession` fails?

Reply:
Also, moving _WD_DeleteSession and _WD_Shutdown prior to the MsgBox commands will shutdown the browser before the user can review the browser window. Was that your intent?

Reply:
> I'm failing to see how this is an improvement. Do you intend to call `_WD_DeleteSession` even when `_WD_CreateSession` fails?

My intention was to close session/browser when  "FireFox Software updater"  occurs.

Now I see my fault that this is imposible  using this:
```AutoIt
		_WD_DeleteSession($sSession)
```
When session is empty string.
Thanks

Reply:
> Also, moving _WD_DeleteSession and _WD_Shutdown prior to the MsgBox commands will shutdown the browser before the user can review the browser window. Was that your intent?


Please check: [_WD_DeleteSession and _WD_Shutdown fixes](https://github.com/Danp2/WebDriver/pull/176/commits/fe942bd82c6df15ce4084d8a396cd52fe2b6454e)

Reply:
That's better, but it still doesn't address this --

>Also, moving _WD_DeleteSession and _WD_Shutdown prior to the MsgBox commands will shutdown the browser before the user can review the browser window. Was that your intent?

Reply:
Please check: [](https://github.com/mlipok)@mlipok
[_WD_DeleteSession and _WD_Shutdown - moved](https://github.com/Danp2/WebDriver/pull/176/commits/d7ab7987b4de02ac61903164dd11ab3eb537161a)